insights/combiners/package_provides_java.py

```diff
@@ -0,0 +1,55 @@
+"""
+Combiner for packages which provide java
+========================================
+
+Combiner for collecting all the PackageProvidesJava objects with two properties, one is running java command
+and the other is package name that provides corresponding java command.
+It provides interfaces to get all the valid such kind of objects and a package that provides certain java command.
+
+Examples:
+    >>> PACKAGE_COMMAND_MATCH_1 = '''/usr/lib/jvm/jre/bin/java java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64'''
+    >>> PACKAGE_COMMAND_MATCH_2 = '''/usr/lib/jvm/java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64/bin/java java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64'''
+    >>> pack1 = PackageProvidesJava(context_wrap(PACKAGE_COMMAND_MATCH_1))
+    >>> pack2 = PackageProvidesJava(context_wrap(PACKAGE_COMMAND_MATCH_2))
+    >>> shared = [{PackageProvidesJavaAll: [pack1, pack2]}]
+    >>> packages = shared[PackageProvidesJavaAll]
+    >>> package = packages[0]
+    >>> package.command
+    '/usr/lib/jvm/jre/bin/java'
+    >>> package.package
+    'java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64'
+    >>> packages.get_package("/usr/lib/jvm/jre/bin/java")
+    'java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64'
+"""
+
+from .. import LegacyItemAccess
+from insights.core.plugins import combiner
+from insights.parsers.package_provides_java import PackageProvidesJava
+
+
+@combiner(PackageProvidesJava)
+class PackageProvidesJavaAll(LegacyItemAccess):
+    """
+    Combiner for collecting all the PackageProvidesJava objects with two properties, one is running java command
+    and the other is package name that provides corresponding java command.
+
+    Attributes:
+        data (list): List of objects containing package name and its providing java command
+    """
+
+    def __init__(self, package_provides_java):
+        packages_list = []
+        for package in package_provides_java:
+            packages_list.append(package)
+        self.data = packages_list
+        super(PackageProvidesJavaAll, self).__init__()
+
+    def __iter__(self):
+        for item in self.data:
+            yield item
+
+    def get_package(self, ps_command):
+        for item in self.data:
```

Comment:
It's necessary to add a corresponding docstring for the new `get_package` interface.